packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -76,7 +84,7 @@ function findFirstIp (str) {
     const type = net.isIP(chunk)
     if (!type) continue
 
-    if (!privateIPMatcher.check(chunk, type === 6 ? 'ipv6' : 'ipv4')) {
+    if (isPublicIp(chunk, type)) {
       // it's public, return it immediately
       result.public = chunk
       break
```

Comment:
maybe we can do `return result` just like in the new function, to keep some consistnecy